pylint/extensions/for_any_all.py

```diff
@@ -36,19 +41,104 @@ def visit_for(self, node: nodes.For) -> None:
             return
 
         if_children = list(node.body[0].get_children())
-        if not len(if_children) == 2:  # The If node has only a comparison and return
-            return
-        if not returns_bool(if_children[1]):
+        if any(isinstance(child, nodes.If) for child in if_children):
+            # an if node within the if-children indicates an elif clause,
+            # suggesting complex logic.
             return
```

Comment:
I don't have a deep comprehension of the problem but intuitively I suppose there could be other children than "if node" that would also be problematic ? Especially since we do not have a number of child's limit anymore. The primer result indicate I'm probably wrong but I don't understand why.

Reply:
The number of child nodes is handled now within `if_statement_returns_bool` which was the original check before this PR. The reason we can't now limit to only two children as we did before is because there are legitimate cases (in tests too) where there are more than 2 children.

The entire logic in this check is very edge-case-y.